catalogue_of_life/Loader.java

```diff
@@ -0,0 +1,242 @@
+    }
+
+    public static Optional<Options> parseCLIOptions(String[] args) {
+        CommandLine commandLine = new CommandLine(new Options());
+        try {
+            CommandLine.ParseResult parseResult = commandLine.parseArgs(args);
+            if (commandLine.isUsageHelpRequested()) {
+                commandLine.usage(commandLine.getOut());
+                return Optional.empty();
+            } else if (commandLine.isVersionHelpRequested()) {
+                commandLine.printVersionHelp(commandLine.getOut());
+                return Optional.empty();
+            } else {
+                return Optional.of(parseResult.asCommandLineList().get(0).getCommand());
+            }
+        } catch (CommandLine.ParameterException ex) {
+            commandLine.getErr().println(ex.getMessage());
+            if (!CommandLine.UnmatchedArgumentException.printSuggestions(ex, commandLine.getErr())) {
+                ex.getCommandLine().usage(commandLine.getErr());
+            }
+            return Optional.empty();
+        }
+    }
+
+    public static Path fetchZip(Path workingDirectory, Options options) throws MalformedURLException {
+        URL url = new URL("https://api.checklistbank.org/dataset/9817/export.zip?format=ColDP");
+
+        Path zipFile = workingDirectory.resolve("coldp.zip");
+        if (!options.skipDownload() && (options.forceDownload() || !Files.exists(zipFile))) {
+            try (InputStream inputStream = url.openStream()) {
+                Files.copy(inputStream, zipFile, StandardCopyOption.REPLACE_EXISTING);
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+        }
+
+        return zipFile;
+    }
+
+    private static void unzip(Path zipFile, Path workingDirectory) {
+        try (ZipInputStream zipIn = new ZipInputStream(Files.newInputStream(zipFile))) {
+            for (ZipEntry zipEntry; (zipEntry = zipIn.getNextEntry()) != null; ) {
+                Path resolvedPath = workingDirectory.resolve(zipEntry.getName());
+                if (zipEntry.isDirectory()) {
+                    Files.createDirectories(resolvedPath);
+                } else {
+                    Files.createDirectories(resolvedPath.getParent());
+                    Files.copy(zipIn, resolvedPath, StandardCopyOption.REPLACE_EXISTING);
+                }
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static void prepareData(Path dataDirectory) throws IOException {
+        TsvParser distributionsParser = beginParsingFile(dataDirectory.resolve("Distribution.tsv"));
+        TsvWriter marineRegionsWriter = createTsvWriter(dataDirectory.resolve("MarineDistribution.tsv"));
+        TsvWriter describedRegionsWriter = createTsvWriter(dataDirectory.resolve("DescribedDistribution.tsv"));
+
+        Record record;
+        Map<String, String> recordValues = new HashMap<>();
```

Comment:
univocity doesn't support writing a Record, so we're forced to do a little switcharoo with a HashMap